### ISSUE TITLE
Not setting MTU on ppp interface

### ISSUE THREAD

**Issue by bdavj** [TotalGriffLock/udmp-jumbo-frames-ppp#3]
``Using a UDM-SE

The ethernet interface MTU gets set to 1508, and the ppp config file updated to mtu 1500, however the link itself (ip link list ppp0) doesnt get set to 1500 MTU.

I have added a script in /etc/ppp/ip-up.d to run the following as a band-aid. Should this be set by the pppd?
```
#!/bin/sh
# based on script from ddclient package

# # These variables are for the use of the scripts run by run-parts
# PPP_IFACE="$1"
# PPP_TTY="$2"
# PPP_SPEED="$3"
# PPP_LOCAL="$4"
# PPP_REMOTE="$5"
# PPP_IPPARAM="$6"

echo "Interface $PPP_IFACE"
if [ "$PPP_IFACE" = "ppp0" ]; then
    ip link set ppp0 mtu 1500
    echo "Set PPP MTU"
fi

```

**Comment by TotalGriffLock** [comment]
Am looking into this - I've noticed this on my own UDM SE as well. It definitely worked when I wrote it. Unfortunately calling ip link set mtu on an existing PPP link doesn't do anything other than update the number that the ip command shows you, if you do some ping tests with the DF bit set you will see your MTU is still 1492. MTU can only be set on PPP links at link negotiation time.

**Comment by bdavj** [comment]
I'm not sure thats the case (or indeed I may well be testing wrong!) - When I change the ppp mtu using ip link I see the "Frag needed and DF set" warning:
![image](https://user-images.githubusercontent.com/5160471/190687094-350d0088-5e77-4487-928b-8780492fe20b.png)

**Comment by bdavj** [comment]
Did you gain any more clarity on this? I’m not convinced my way is working consistently either, nor whether my ping test is accurate. I’ve used a couple of test tools online and get different results to my ping test depending on the order in which I do things on the UDM

**Comment by TotalGriffLock** [comment]
I did, I think. From what I can work out in my testing, if you set the MTU in the UDM shell once the link is established, it sets it for your end of the link, so the UDM starts being able to transmit frames of 1500 bytes, so your local tests from behind the UDM will appear to work. What it doesn't do is change it at your ISPs end of the link, so the packets that come to you from the internet are still limited, so that means online tests of your MTU such as http://www.letmecheck.it/mtu-test.php fail even with the UDM showing the correct MTU.

In other news I think I've figured out why the script was misbehaving in your case, and have refactored it a little bit. I have tested it on my UDM-SE and it works great, but I still have an outstanding issue where I'm relying on bash which doesn't exist on the UDM-Pro. Just needs an if statement in it to determine the platform.
